src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -603,6 +653,8 @@ void CallGenerator::do_late_inline_helper() {
     C->remove_macro_node(call);
   }
 
+  replace_box_to_scalar(call, callprojs.resproj);
```

Comment:
Should this be guarded by `C->eliminate_boxing()`?

Reply:
Exactly. I will change that.